tsconfig.json

```diff
@@ -12,6 +12,7 @@
     "checkJs": false,
     "allowSyntheticDefaultImports": true,
     "resolveJsonModule": true,
+    "jsx": "preserve",
```

Comment:
Huh. Why is this necessary?

Reply:
oh - oops, that's one of the flags used when compiling Vue SFCs from TS. earlier i experimented with doing some incremental TS migration by building the new components with it. after running into trouble with the TSC segfault, i set aside that idea, but evidently i forgot to clean up after.

come to think of it, that might be a good idea to revisit now that ive patched the segfault, but doing that for this + the buttons could be its own PR.